File: Employee_Directory/Models/Emp.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Web;

namespace Employee_Directory.Models
{
    public class Emp
    {
        public class Details_Param

        {
            public string Emp_ID { get; set; }
            
        }
        public class Details_Paramss

        {
            public string Location { get; set; }

        }
        public class Status1
      
        {
            public string Emp_ID { get; set; }
            public string Emp_Name { get; set; }
            public string Emp_Address { get; set; }
            public string Emp_Desg { get; set; }
            public string Emp_totalExp { get; set; }
            public string Emp_InHouse_Exp { get; set; }
            public string Emp_Team { get; set; }
            public string Emp_Rp_Manager { get; set; }
            public string Emp_Days_Present { get; set; }
            public string Emp_Casucal { get; set; }
            public string Emp_Silck { get; set; }
            public string Emp_Date { get; set; }
            public string Emp_Leave_Type { get; set; }
            public string Emp_Count { get; set; }
            public string Emp_Status { get; set; }
            public string Emp_Probation { get; set; }
            public string Emp_Message { get; set; }
            public string Emp_Acessable { get; set; }
            public string Emp_Payable { get; set; }
            public string Emp_Dep { get; set; }
            public string Emp_ImageURL { get; set; }


        }

        public class Loc

        {
            public string Name { get; set; }
            public string ID { get; set; }
        }

            private SqlConnection con;
        private void connection()
        {
            string constring = ConfigurationManager.ConnectionStrings["StudentConn"].ToString();
            con = new SqlConnection(constring);

        }

        public List<Status1> Emp_Details(Details_Param U)
        {
            List<Status1> resultList = new List<Status1>();

            connection();
            using (SqlCommand Cmd = new SqlCommand("Emp_Details", con))
            {
                Cmd.CommandType = CommandType.StoredProcedure;
                Cmd.Parameters.AddWithValue("@Emp_ID", U.Emp_ID);
                con.Open();
                using (SqlDataReader reader = Cmd.ExecuteReader())
                {
                    
                    while (reader.HasRows && reader.Read())
                    {
                        Status1 result = new Status1
                        {
                            Emp_ID = reader["Emp_ID"].ToString(),
                            Emp_Name = reader["Emp_Name"].ToString(),
                            Emp_Address = reader["Emp_Address"].ToString(),
                            Emp_Desg = reader["Emp_Desg"].ToString(),
                            Emp_totalExp = reader["Emp_totalExp"].ToString(),
                            Emp_InHouse_Exp = reader["Emp_InHouse_Exp"].ToString(),
                            Emp_Team = reader["Emp_Team"].ToString(),
                            Emp_Rp_Manager = reader["Emp_Rp_Manager"].ToString(),
                            Emp_Days_Present = reader["Emp_Days_Present"].ToString(),
                            Emp_Casucal = reader["Emp_Casucal"].ToString(),
                            Emp_Silck = reader["Emp_Silck"].ToString(),
                            Emp_Date = reader["Emp_Date"].ToString(),
                            Emp_Leave_Type = reader["Emp_Leave_Type"].ToString(),
                            Emp_Count = reader["Emp_Count"].ToString(),
                            Emp_Status = reader["Emp_Status"].ToString(),
                            Emp_Probation = reader["Emp_Probation"].ToString(),
                            Emp_Message = reader["Emp_Message"].ToString(),
                            Emp_Acessable = reader["Emp_Acessable"].ToString(),
                            Emp_Payable = reader["Emp_Payable"].ToString(),
                            Emp_Dep = reader["Emp_Dep"].ToString(),
                            Emp_ImageURL = reader["Emp_ImageURL"].ToString()
                        };

                        resultList.Add(result);
                    }
                }
                con.Close();
            }

            return resultList;
        }

        public List<Loc> EmpLocation()
        {
            List<Loc> resultList = new List<Loc>();

            connection();
            using (SqlCommand Cmd = new SqlCommand("Emp_Loc", con))
            {
                con.Open();
                using (SqlDataReader reader = Cmd.ExecuteReader())
                {
                    while (reader.HasRows && reader.Read())
                    {
                        Loc result = new Loc
                        {
                            Name = reader["Name"].ToString(),
                            ID = reader["ID"].ToString(),

                        };

                        resultList.Add(result);
                    }
                }
                con.Close();
            }

            return resultList;
        }

        public List<Loc> EmpDep()
        {
            List<Loc> resultList = new List<Loc>();

            connection();
            using (SqlCommand Cmd = new SqlCommand("Emp_Dep_Sp", con))
            {
                con.Open();
                using (SqlDataReader reader = Cmd.ExecuteReader())
                {
                    while (reader.HasRows && reader.Read())
                    {
                        Loc result = new Loc
                        {
                            Name = reader["Name"].ToString(),
                            ID = reader["ID"].ToString(),

                        };

                        resultList.Add(result);
                    }
                }
                con.Close();
            }

            return resultList;
        }

        public List<Status1> Emp_DetailsCall(Details_Paramss U)
        {
            List<Status1> resultList = new List<Status1>();

            connection();
            using (SqlCommand Cmd = new SqlCommand("Fill_LocTion", con))
            {
                Cmd.CommandType = CommandType.StoredProcedure;
                Cmd.Parameters.AddWithValue("@Location", U.Location);
                con.Open();
                using (SqlDataReader reader = Cmd.ExecuteReader())
                {

                    while (reader.HasRows && reader.Read())
                    {
                        Status1 result = new Status1
                        {
                            Emp_ID = reader["Emp_ID"].ToString(),
                            Emp_Name = reader["Emp_Name"].ToString(),
                            Emp_Address = reader["Emp_Address"].ToString(),
                            Emp_Desg = reader["Emp_Desg"].ToString(),
                            Emp_totalExp = reader["Emp_totalExp"].ToString(),
                            Emp_InHouse_Exp = reader["Emp_InHouse_Exp"].ToString(),
                            Emp_Team = reader["Emp_Team"].ToString(),
                            Emp_Rp_Manager = reader["Emp_Rp_Manager"].ToString(),
                            Emp_Days_Present = reader["Emp_Days_Present"].ToString(),
                            Emp_Casucal = reader["Emp_Casucal"].ToString(),
                            Emp_Silck = reader["Emp_Silck"].ToString(),
                            Emp_Date = reader["Emp_Date"].ToString(),
                            Emp_Leave_Type = reader["Emp_Leave_Type"].ToString(),
                            Emp_Count = reader["Emp_Count"].ToString(),
                            Emp_Status = reader["Emp_Status"].ToString(),
                            Emp_Probation = reader["Emp_Probation"].ToString(),
                            Emp_Message = reader["Emp_Message"].ToString(),
                            Emp_Acessable = reader["Emp_Acessable"].ToString(),
                            Emp_Payable = reader["Emp_Payable"].ToString(),
                            Emp_Dep = reader["Emp_Dep"].ToString(),
                            Emp_ImageURL = reader["Emp_ImageURL"].ToString()
                        };

                        resultList.Add(result);
                    }
                }
                con.Close();
            }

            return resultList;
        }
    }
}

File: Employee_Directory/JS/Emp.js
$(document).ready(function () {
    Fill_Passing_Type('location', '');
    Fill_Dep_Type('department', '');
    FetchEmpDetails();
    $('#employeeModal .close').on('click', function () {
        $('#employeeModal').hide();
    });
   
});
function FetchEmpDetails() {
    var Emp_ID = "0";
    let data = {
        Emp_ID: Emp_ID
    };

    $.ajax({
        cache: false,
        type: "POST",
        url: "/Emp/EmpDetails",
        data: JSON.stringify(data),
        contentType: "application/json; charset=utf-8",
        dataType: "json",
        async: true,
        success: function (response) {
            var empDetails = response;

            $('#EmpDetails').empty();
            if (Array.isArray(empDetails)) {
                $.each(empDetails, function (index, emp) {
                    var imageUrl = emp.Emp_ImageURL;

                    $('#EmpDetails').append(
                        '<div class="emp-detail" data-emp-id="' + emp.Emp_ID + '">' +
                        '<img src="' + imageUrl + '" alt="Profile Picture" class="profile-pic">' +
                        '<h4>' + emp.Emp_Name + '</h4>' +
                        '<p>' + emp.Emp_Desg + '</p>' +
                        '<p>' + emp.Emp_Dep + '</p>' +
                        '<p>' + emp.Emp_ID + '</p>' +
                        '</div>'
                    );
                });

              
                $('.emp-detail').on('click', function () {
                    var empId = $(this).data('emp-id');
                    OpenEmpDetailModal(empId);
                });
            } else {
                $('#EmpDetails').append('<p>Unexpected data format received.</p>');
            }
        },
        complete: function () {
          
        },
        error: function () {
            $('#EmpDetails').append('<p>Error loading employee details. Please try again later.</p>');
        }
    });
}

function OpenEmpDetailModal(empId) {
    let data = {
        Emp_ID: empId
    };
    $.ajax({
        cache: false,
        type: "POST",
        url: "/Emp/EmpDetails",
        data: JSON.stringify(data),
        contentType: "application/json; charset=utf-8",
        dataType: "json",
        async: true,
        success: function (response) {
            console.log('Employee Details:', response);
            if (response && response.length > 0) {
                var details = response[0];

                // Update profile picture
                $('#employeeModal .profile-pic').attr('src', details.Emp_ImageURL || '');

                // Clear existing content in user-info
                $('#employeeModal .user-info').empty();

                // Append new content
                $('#employeeModal .user-info').append(`
                    <p><strong>ID:</strong> ${details.Emp_ID || 'N/A'}</p>
                    <p><strong>Designation:</strong> ${details.Emp_Desg || 'N/A'}</p>
                    <p><strong>Address:</strong> ${details.Emp_Address || 'N/A'}</p>
                    <p><strong>Total Experience:</strong> ${details.Emp_totalExp || 'N/A'}</p>
                    <p><strong>In-House Experience:</strong> ${details.Emp_InHouse_Exp || 'N/A'}</p>
                    <p><strong>Team:</strong> ${details.Emp_Team || 'N/A'}</p>
                    <p><strong>Reporting Manager:</strong> ${details.Emp_Rp_Manager || 'N/A'}</p>
                    <p><strong>Days Present:</strong> ${details.Emp_Days_Present || 'N/A'}</p>
                    <p><strong>Casual Leaves:</strong> ${details.Emp_Casucal || 'N/A'}</p>
                    <p><strong>Sick Leaves:</strong> ${details.Emp_Silck || 'N/A'}</p>
                    <p><strong>Date:</strong> ${details.Emp_Date || 'N/A'}</p>
                    <p><strong>Leave Type:</strong> ${details.Emp_Leave_Type || 'N/A'}</p>
                    <p><strong>Count:</strong> ${details.Emp_Count || 'N/A'}</p>
                    <p><strong>Status:</strong> ${details.Emp_Status || 'N/A'}</p>
                    <p><strong>Probation:</strong> ${details.Emp_Probation || 'N/A'}</p>
                    <p><strong>Message:</strong> ${details.Emp_Message || 'N/A'}</p>
                    <p><strong>Accessible:</strong> ${details.Emp_Acessable || 'N/A'}</p>
                    <p><strong>Payable:</strong> ${details.Emp_Payable || 'N/A'}</p>
                    <p><strong>Department:</strong> ${details.Emp_Dep || 'N/A'}</p>

                `);

                // Show the modal
                $('#employeeModal').show();
            } else {
                alert('No details found for this employee.');
            }
        },
        error: function (xhr, status, error) {
            console.error('AJAX Error:', status, error);
            alert('Error loading employee details. Please try again later.');
        }
    });
}



  

function Fill_Passing_Type(CmbBx) {

    let DefaultType = '';



    let data =
    {

        DefaultType: DefaultType
    }
    $.ajax({
        cache: false,
        type: "POST",
        url: "/Emp/EmpLocation",
        data: JSON.stringify(data),
        contentType: "application/json; charset=utf-8",
        dataType: "json",
        async: false,

        success: function (data) {
            $(`#${CmbBx}`).empty();
            for (var i in data) {
                var Opt = new Option(data[i].Name, data[i].ID);
                $(`#${CmbBx}`).append(Opt);
            }

        },
        complete: function (data) {
            $("#overlay").hide();
        },
        error: function (data, success, error) {
            alert(error);
        }
    });
    }

function Fill_Dep_Type(CmbBx) {

    let DefaultType = '';



    let data =
    {

        DefaultType: DefaultType
    }
    $.ajax({
        cache: false,
        type: "POST",
        url: "/Emp/EmpDep",
        data: JSON.stringify(data),
        contentType: "application/json; charset=utf-8",
        dataType: "json",
        async: false,

        success: function (data) {
            $(`#${CmbBx}`).empty();
            for (var i in data) {
                var Opt = new Option(data[i].Name, data[i].ID);
                $(`#${CmbBx}`).append(Opt);
            }

        },
        complete: function (data) {
            $("#overlay").hide();
        },
        error: function (data, success, error) {
            alert(error);
        }
    });
}